### PR TITLE
DisputableVoting: cosmetic polish

### PR DESCRIPTION
**Pull request by sohkai**
Similar to Agreements; this is just a bunch of polish. As before, easiest to review commit by commit; the last one (9b9d0a3) is one that just reorders/regroups the functions but adds no other changes.

In terms of ordering, it was a bit arbitrary at times, but I thought this was most natural:

- Settings
- Vote (creation, voting, "extension lifecycle")
- Voters (voter states, etc.)
- Misc.

More comments inline.

### CODE REVIEW

File: apps/voting-disputable/arapp.json [hunk 1]
```diff
@@ -72,7 +72,7 @@
     },
     {
       "name": "Modify quiet ending configuration",
-      "id": "MODIFY_QUIET_ENDING_CONFIGURATION",
+      "id": "MODIFY_QUIET_ENDING_ROLE",
```
Comment (sohkai) [inline apps/voting-disputable/arapp.json:75]: I don't think this will have any breaking changes for us anywhere yet, but good note that we always suffix roles with `_ROLE` :)

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 2]
```diff
@@ -40,23 +40,23 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
-    string private constant ERROR_CHANGE_QUORUM_PCTS = "VOTING_CHANGE_QUORUM_PCTS";
-    string private constant ERROR_CHANGE_SUPPORT_PCTS = "VOTING_CHANGE_SUPPORT_PCTS";
-    string private constant ERROR_CHANGE_SUPPORT_TOO_BIG = "VOTING_CHANGE_SUPP_TOO_BIG";
-    string private constant ERROR_INVALID_OVERRULE_WINDOW = "VOTING_INVALID_OVERRULE_WINDOW";
-    string private constant ERROR_INVALID_QUIET_ENDING_PERIOD = "VOTING_INVALID_QUIET_ENDING_PERI";
-    string private constant ERROR_DELEGATES_EXCEEDS_MAX_LEN = "VOTING_DELEGATES_EXCEEDS_MAX_LEN";
     string private constant ERROR_SETTING_DOES_NOT_EXIST = "VOTING_SETTING_DOES_NOT_EXIST";
+    string private constant ERROR_CHANGE_QUORUM_TOO_BIG = "VOTING_CHANGE_QUORUM_TOO_BIG";
+    string private constant ERROR_CHANGE_SUPPORT_TOO_SMALL = "VOTING_CHANGE_SUPPORT_TOO_SMALL";
+    string private constant ERROR_CHANGE_SUPPORT_TOO_BIG = "VOTING_CHANGE_SUPPORT_TOO_BIG";
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol:46]: Just updated these to be a bit more descriptive of why they're an error

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 3]
```diff
@@ -40,23 +40,23 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
-    string private constant ERROR_CANNOT_VOTE = "VOTING_CANNOT_VOTE";
-    string private constant ERROR_CANNOT_EXECUTE = "VOTING_CANNOT_EXECUTE";
     string private constant ERROR_CANNOT_FORWARD = "VOTING_CANNOT_FORWARD";
-    string private constant ERROR_CANNOT_PAUSE_VOTE = "VOTING_CANNOT_PAUSE_VOTE";
     string private constant ERROR_NO_VOTING_POWER = "VOTING_NO_VOTING_POWER";
-    string private constant ERROR_VOTE_NOT_PAUSED = "VOTING_VOTE_NOT_PAUSED";
+    string private constant ERROR_CANNOT_VOTE = "VOTING_CANNOT_VOTE";
     string private constant ERROR_NOT_REPRESENTATIVE = "VOTING_NOT_REPRESENTATIVE";
-    string private constant ERROR_CANNOT_DELEGATE_VOTE = "VOTING_CANNOT_DELEGATE_VOTE";
+    string private constant ERROR_PAST_REPRESENTATIVE_VOTING_WINDOW = "VOTING_PAST_REP_VOTING_WINDOW";
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol:55]: And this one too

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 4]
```diff
@@ -62,52 +62,67 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
+
+        // Duration before the end of a vote to detect non-quiet endings
+        // Must be <= voteTime
         uint64 quietEndingPeriod;
+
+        // Duration to extend a vote in case of non-quiet ending
         uint64 quietEndingExtension;
+
+        // Duration to wait before a passed vote can be executed
+        uint64 executionDelay;
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol:92]: I thought it'd be useful to describe these settings a bit more, although we will likely want to revisit these descriptions once we re-think these "windows" to be based on the start time rather than end time.

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 5]
```diff
@@ -733,160 +759,169 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
     function _canDelegateVote(Vote storage _vote) internal view returns (bool) {
         return _isActive(_vote) && !_withinOverruleWindow(_vote);
     }
 
     /**
-    * @dev Internal function to check if a voter can participate in a vote
-    *      It assumes the pointer to the vote is valid
-    * @return True if the given voter can participate a certain vote
+    * @dev Tell if a voter can participate in a vote.
+    *      Note that a voter cannot change their vote once cast, except for the principal voter
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol]: I've stuck with the following terms for voters:

- Principal, delegated voter: always refer to the original token holder
- Representative: a delegate voting for others

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 6]
```diff
@@ -733,160 +759,169 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
     function _finalVoteEndDate(Vote storage _vote) internal view returns (uint64) {
         uint64 endDateAfterPause = _originalVoteEndDate(_vote).add(_vote.pauseDuration);
         return endDateAfterPause.add(_vote.quietEndingExtendedSeconds);
     }
 
     /**
-    * @dev Internal function to tell whether a vote was flipped or not based on its snapshot support
-    *      It assumes the pointer to the vote is valid
+    * @dev Tell if a vote was flipped in its most recent quiet ending period
+    *      This function assumes that it will only be called after the most recent quiet ending period has already ended
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol]: I thought this note was important to highlight to make sure I'm understanding how this is meant to be used.

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 7]
```diff
@@ -654,18 +679,20 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
...
     }
 
     /**
-    * @dev Internal function for a representative to cast a vote on behalf of many voters. It assumes the queried vote exists.
+    * @dev Cast votes on behalf of delegated voters as a representative.
     */
     function _voteOnBehalfOf(uint256 _voteId, bool _supports, address[] _voters) internal {
         Vote storage vote_ = _getVote(_voteId);
+        // Note that the period for representatives to vote can never go into a quiet ending
+        // extension, and so we don't need to check other timing-based pre-conditions
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol:708]: Similarly, this one as well—voting as a representative has less "time-based" checks than when voting normally and I believe this explains why

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 8]
```diff
@@ -287,8 +302,8 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
         vote_.status = VoteStatus.Executed;
         _closeDisputableAction(vote_.actionId);
 
-        // Add Agreement to blacklist to disallow the stored EVMScript from directly calling the
-        // linked Agreement from this app's context (e.g. maliciously creating a new action)
+        // Add attached Agreement to blacklist to disallow the stored EVMScript from directly calling
+        // the Agreement from this app's context (e.g. maliciously creating a new action)
```
Comment (sohkai) [inline apps/voting-disputable/contracts/DisputableVoting.sol:285]:
```suggestion
        // the Agreement from this app's context (e.g. maliciously close a different action)
```

I think this is actually a better example than the old one :)

File: apps/agreement/contracts/Agreement.sol [hunk 9]
```diff
@@ -124,7 +124,7 @@ contract Agreement is ILockManager, IAgreement, IArbitrable, IACLOracle, AragonA
```
Comment (sohkai) [inline apps/agreement/contracts/Agreement.sol:124]: Let me know if I should pull out e73a814678b6705ef656fb5f50d939a0c71337d1 into a different PR—it's quite small so I didn't bother too much.

File: apps/agreement/contracts/Agreement.sol [hunk 10]
```diff
@@ -1178,7 +1178,7 @@ contract Agreement is ILockManager, IAgreement, IArbitrable, IACLOracle, AragonA
     * @param _collateralToken Address of the ERC20 token to be used for collateral
     * @param _actionAmount Amount of collateral tokens that will be locked every time an action is submitted
     * @param _challengeAmount Amount of collateral tokens that will be locked every time an action is challenged
-    * @param _challengeDuration Challenge duration in seconds, during which the submitter can raise a dispute
+    * @param _challengeDuration Challenge duration, during which the submitter can raise a dispute
```
Comment (sohkai) [inline apps/agreement/contracts/Agreement.sol:1181]: Thought it was pointless to describe it in "seconds"; we should assume all Ethereum durations are seconds :)

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 11]
```diff
@@ -1065,7 +1123,9 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
     }
 
     /**
-    * @dev Private function to tell the voter state for a support option
+    * @dev Translate a voter's support into a voter state
+    * @param _supports Whether voter supports the vote
+    @ return Voter state, as an enum
```
Comment (facuspagnuolo) [inline apps/voting-disputable/contracts/DisputableVoting.sol:1128]:
```suggestion
    * @return Voter state, as an enum
```

File: apps/agreement/contracts/Agreement.sol (hunk 9 above)
Comment (facuspagnuolo) [reply apps/agreement/contracts/Agreement.sol:124]: Totally fine :)

File: apps/voting-disputable/contracts/DisputableVoting.sol (hunk 5 above)
Comment (facuspagnuolo) [reply apps/voting-disputable/contracts/DisputableVoting.sol]: I think principal and representative are okay :)
